tests/integration/integration-test.py

```diff
@@ -79,93 +85,166 @@ def test_get_notes_query_by_student_id(self, endpoint='/notes'):
                 self.assertIn(q, actual_note)
 
-    # /notes?studentId=111111111&sources=
+    def query_sort_keys(self, student_id, endpoint='/notes'):
+        sort_keys = ['lastModified', 'source', 'permissions', 'contextType']
+        for sort_key in sort_keys:
+            params = {'studentId': student_id, 'sortKey': sort_key}
+            response = self.get_response(params)
+            # Validating each sort_key
+            if sort_key == 'lastModified':
+                self.check_time_sort(response.json()['data'])
+            elif sort_key == 'source' or sort_key == 'permissions':
+                self.check_alphabetical_sort(response.json()['data'], sort_key)
+            elif sort_key == 'contextType':
+                self.check_context_type_sort(response.json()['data'])
+
+        # invalid sortKey tests returns 400
+        invalid_sources = [" ", "lastmodified", "random", "contexttype"]
+        invalid_source = random.choice(invalid_sources)
+        params = {'studentId': student_id, 'sortKey': invalid_source}
+        response = utils.make_request(self, endpoint, 400, params=params)
+        schema = utils.get_resource_schema(self, 'Error')
+        utils.check_schema(self, response, schema)
+
+    def check_time_sort(self, response):
+        try:
+            last_modified = response[0]['attributes']['lastModified']
+            greater_element = datetime.strptime(last_modified,
+                                                '%Y-%m-%dT%H:%M:%S.%f%z')
+            for resource in response:
+                current_element = resource['attributes']['lastModified']
+                current_element = datetime.strptime(current_element,
+                                                    '%Y-%m-%dT%H:%M:%S.%f%z')
+                self.assertGreaterEqual(greater_element, current_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
+    def check_alphabetical_sort(self, response, sort_key):
+        try:
+            greater_element = response[0]['attributes'][sort_key]
+            for resource in response:
+                current_element = resource['attributes'][sort_key]
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
+    def check_context_type_sort(self, response):
+        try:
+            attributes = response[0]['attributes']
+            greater_element = attributes['context']['contextType']
+            for resource in response:
+                attributes = resource['attributes']
+                current_element = attributes['context']['contextType']
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
```

Comment:
You shouldn't handle index error like this. If we have more than one places in the try block which is trying to access an item by its index, this except block will catch those index error no matter what and then run `self.assertEqual(response, [])`. It can potentially "miss-succeed" the test case if we do actually run into some errors without catching them correctly.